sootup.core/src/main/java/sootup/core/graph/DominanceFinder.java

```diff
@@ -49,49 +47,58 @@ public enum AnalysisDirection {
       List<? extends BasicBlock<?>> getPredecessors(BasicBlock<?> block) {
         return block.getSuccessors();
       }
+
+      @Nonnull
+      @Override
+      List<BasicBlock<?>> getSortedBlocks(StmtGraph<?> blockGraph) {
+        return Collections.unmodifiableList(new BackwardsStmtGraph(blockGraph).getBlocksSorted());
+      }
     },
     FORWARD {
       @Override
       @Nonnull
       List<? extends BasicBlock<?>> getPredecessors(BasicBlock<?> block) {
         return block.getPredecessors();
       }
+
+      @Nonnull
+      @Override
+      List<BasicBlock<?>> getSortedBlocks(StmtGraph<?> blockGraph) {
+        return Collections.unmodifiableList(new ForwardingStmtGraph(blockGraph).getBlocksSorted());
```

Comment:
I think we can move the forwards implementation up from `ForwardingStmtGraph` to the StmtGraph as the ForwardingStmtGraph refers not to the direction but that its proxying the calls to the backing StmtGraph object.

Reply:
Got it! I found the  `ForwardingStmtGraph` is  proxying backingGraph which is actually `MutableBlockStmtGraph`, I temporarily moved the impletation to `MutableBlockStmtGraph`, which would make `BlockIterator` temporarily unused.

```java
  // MutableBlockStmtGraph.java
  @Nonnull
  public List<? extends BasicBlock<?>> getBlocksSorted() {
    return ReversePostOrderBlockTraversal.getBlocksSorted(this);
  }
```

Or should I solve it by moveing the impl to `StmtGraph` and use following code in `ForwardStmtGraph`

```java
      //ForwardStmtGraph.java
      @Nonnull
      @Override
      List<BasicBlock<?>> getSortedBlocks(StmtGraph<?> blockGraph) {
        return super.getSortedBlocks(blockGraph);  // find StmtGraph
      }
```